### PR TITLE
Refactor types for plugin-ember-octane

### PR DESCRIPTION
**Pull request by lbdm44**
* Move custom types into a `types` directory.
* Remove as many custom types for Ember Template Lint as possible
* Refactor code to use ESLint's report/results API as much as possible.

### CODE REVIEW

File: packages/plugin-ember-octane/src/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export enum TemplateLintMigrationType {
+  AngleBrackets = 'angle-brackets',
+  NamedArgs = 'named-args',
+  OwnProperties = 'own-properties',
+  UseModifiers = 'use-modifiers',
+}
+
+interface CompetionInfo {
```
Comment (scalvert) [inline packages/plugin-ember-octane/src/types/index.ts]:
```suggestion
interface CompletionInfo {
```

File: packages/plugin-ember-octane/src/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  total: number;
+  completed: number;
+  percentage: string;
+}
+
+export interface MigrationInfo {
+  completionInfo: CompetionInfo;
```
Comment (scalvert) [inline packages/plugin-ember-octane/src/types/index.ts]:
```suggestion
  completionInfo: CompletionInfo;
```

File: packages/plugin-ember-octane/src/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface MigrationRuleConfig {
+  fileMatchers: RegExp[];
+  name: string;
+  rules: string[];
+}
+
+export interface EmberTemplateLintMessage {
+  rule: string;
+  severity: 0 | 1 | 2;
```
Comment (scalvert) [inline packages/plugin-ember-octane/src/types/index.ts]: Would this be better represented as an enum, or no?

File: packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts [hunk 4]
```diff
@@ -2,11 +2,10 @@ import { CLIEngine } from 'eslint';
 import { stdout } from '@checkup/test-helpers';
 import { OctaneMigrationStatusTask } from '../../src/tasks';
 import { OctaneMigrationStatusTaskResult } from '../../src/results';
-import { EmberTemplateLintReport } from '../../src/tasks/octane-migration-status-task';
 
 describe('octane-migration-status-task-result', () => {
   let sampleESLintReport: CLIEngine.LintReport;
-  let sampleTemplateLintReport: EmberTemplateLintReport;
+  let sampleTemplateLintReport: CLIEngine.LintReport;
```
Comment (scalvert) [inline packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts]: Did you just make the template lint report output conform to the ESLint format?

Comment (lbdm44) [reply packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts]: Yes, yes I did, it only required transforming the output's `rule` => `ruleId`

File: packages/plugin-ember-octane/src/types/index.ts (hunk 3 above)
Comment (lbdm44) [reply packages/plugin-ember-octane/src/types/index.ts]: in ESLint the severity is a type with either of those three values, I will go ahead and mirror their type defs

File: packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts (hunk 4 above)
Comment (scalvert) [reply packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts]: Ya I totally understand the intent. I do feel this is a bit confusing to readers though, as we're using types from one library for another. It feels a bit like if I were borrowing your underwear...

The fact that I had to validate this with you means it's likely to be confusing to other readers.

Can we still adhere to the same structure, but not necessarily use the exact same types?

Comment (lbdm44) [reply packages/plugin-ember-octane/__tests__/results/octane-migration-status-tast-result-test.ts]: Discussed offline, long term we are going to publish types from Ember Template Lint, for now I am refactoring the types into their own def file so we could eventually swap them out with those published types.

**Pull request merged**